NPE for invalid client Yaml

With 5.0-SNAPSHOT (20210712) if I do

```
new YamlClientConfigBuilder("application.yml").build();
```

And the YAML file contains no `hazelcast-client:` ssection I get an NPE instead of a helpful diagnostic

```
Caused by: java.lang.NullPointerException
	at com.hazelcast.internal.config.YamlConfigSchemaValidator.lambda$validate$0(YamlConfigSchemaValidator.java:72) ~[hazelcast-enterprise-5.0-SNAPSHOT.jar!/:5.0-SNAPSHOT]
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$5.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$5.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline.count(Unknown Source) ~[?:?]
	at com.hazelcast.internal.config.YamlConfigSchemaValidator.validate(YamlConfigSchemaValidator.java:73) ~[hazelcast-enterprise-5.0-SNAPSHOT.jar!/:5.0-SNAPSHOT]
	at com.hazelcast.client.config.YamlClientConfigBuilder.parseAndBuildConfig(YamlClientConfigBuilder.java:160) ~[hazelcast-enterprise-5.0-SNAPSHOT.jar!/:5.0-SNAPSHOT]
	at com.hazelcast.client.config.YamlClientConfigBuilder.build(YamlClientConfigBuilder.java:132) ~[hazelcast-enterprise-5.0-SNAPSHOT.jar!/:5.0-SNAPSHOT]
	at com.hazelcast.client.config.YamlClientConfigBuilder.build(YamlClientConfigBuilder.java:120) ~[hazelcast-enterprise-5.0-SNAPSHOT.jar!/:5.0-SNAPSHOT]
	at com.hazelcast.client.config.YamlClientConfigBuilder.build(YamlClientConfigBuilder.java:115) ~[hazelcast-enterprise-5.0-SNAPSHOT.jar!/:5.0-SNAPSHOT]
```

--- a/hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java
@@ -71,8 +71,9 @@ public class YamlConfigSchemaValidator {
             // this could be expressed in the schema as well, but that would make all the schema validation errors much harder
             // to read, so it is better to implement it here as a semantic check
             long definedRootNodeCount = PERMITTED_ROOT_NODES.stream()
-                    .filter(rootNodeName -> rootNode.child(rootNodeName) != null)
+                    .filter(rootNodeName -> rootNode != null && rootNode.child(rootNodeName) != null)
                     .count();
+
             if (definedRootNodeCount != 1) {
                 throw new SchemaViolationConfigurationException(
                         "exactly one of [hazelcast], [hazelcast-client] and [hazelcast-client-failover] should be present in the"

--- a/hazelcast/src/test/java/com/hazelcast/config/YamlConfigSchemaValidatorTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/config/YamlConfigSchemaValidatorTest.java
@@ -81,4 +81,9 @@ public class YamlConfigSchemaValidatorTest {
     public void emptyObject() {
         new YamlConfigBuilder(getClass().getResourceAsStream("/com/hazelcast/config/empty-object.yaml")).build();
     }
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void witoutRootObject() {
+        new YamlConfigBuilder(getClass().getResourceAsStream("/com/hazelcast/config/without-root-object.yaml")).build();
+    }
 }

--- a/hazelcast/src/test/resources/com/hazelcast/config/without-root-object.yaml
+++ b/hazelcast/src/test/resources/com/hazelcast/config/without-root-object.yaml
@@ -1,0 +1,3 @@
+fakeroot:
+  metrics:
+    enabled: false

